internal/provider/atlas_schema_data_source.go

```diff
@@ -0,0 +1,129 @@
+package provider
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"hash/fnv"
+
+	"ariga.io/atlas/sql/schema"
+	"ariga.io/atlas/sql/sqlclient"
+	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type (
+	// AtlasSchemaDataSource defines the data source implementation.
+	AtlasSchemaDataSource struct{}
+	// AtlasSchemaDataSourceModel describes the data source data model.
+	AtlasSchemaDataSourceModel struct {
+		DevURL types.String `tfsdk:"dev_db_url"`
+		Src    types.String `tfsdk:"src"`
+		HCL    types.String `tfsdk:"hcl"`
+		Id     types.String `tfsdk:"id"`
```

Comment:
```suggestion
		ID     types.String `tfsdk:"id"`
```

Any reason not to use the Go standard?